test/unit/auth.js

```diff
@@ -113,4 +115,224 @@ describe('DropboxAuth', () => {
       }
     });
   });
+
+  describe('clientSecret', () => {
+    it('can be set in the constructor', () => {
+      const dbx = new Dropbox({ clientSecret: 'foo' });
+      chai.assert.equal(dbx.auth.getClientSecret(), 'foo');
+    });
+
+    it('is undefined if not set in constructor', () => {
+      const dbx = new Dropbox();
+      chai.assert.equal(dbx.auth.getClientSecret(), undefined);
+    });
+
+    it('can be set after being instantiated', () => {
+      const dbx = new Dropbox();
+      dbx.auth.setClientSecret('foo');
+      chai.assert.equal(dbx.auth.getClientSecret(), 'foo');
+    });
+  });
+
+  describe('refreshToken', () => {
+    it('can be set in the constructor', () => {
+      const dbxAuth = new DropboxAuth({ refreshToken: 'foo' });
+      chai.assert.equal(dbxAuth.getRefreshToken(), 'foo');
+    });
+
+    it('is undefined if not set in constructor', () => {
+      const dbxAuth = new DropboxAuth();
+      chai.assert.equal(dbxAuth.getRefreshToken(), undefined);
+    });
+
+    it('can be set after being instantiated', () => {
+      const dbxAuth = new DropboxAuth();
+      dbxAuth.setRefreshToken('foo');
+      chai.assert.equal(dbxAuth.getRefreshToken(), 'foo');
+    });
+  });
+
+  describe('AccessTokenExpiresAt', () => {
+    it('can be set in the constructor', () => {
+      const date = new Date();
```

Comment:
Can we try setting this to an explicit time instead of using the default date? I don't know how the internal date object works but if somehow internally is just constructing `new Date()` this test would still work right?

Reply:
Yea it basically just constructs a new date for whatever the date is at time of execution but we can make it explicit.